surefire-providers/surefire-junit-platform/src/main/java/org/apache/maven/surefire/junitplatform/JUnitPlatformProvider.java

```diff
@@ -104,22 +104,29 @@ public JUnitPlatformProvider( ProviderParameters parameters )
     public RunResult invoke( Object forkTestSet )
                     throws TestSetFailedException, ReporterException
     {
+        ReporterFactory reporterFactory = parameters.getReporterFactory();
```

Comment:
All this PR has the same call sequences as before.
I still do not see logical sense for this change because there is no change in call sequences.
What root cause you found?

Reply:
@sormuras 
I found the root cause. It was caused by calling method `scanClasspath()` before `startCapture`.
Change your commit and use `try-finally` in the method `invoke` and use the code within `finally` block as before.

Reply:
Good catch!

Know I understand the reason for that strange try/finally. Can we add a comment?